### PR TITLE
[BUGFIX] clean up unique alias for all deleted job records

### PR DESCRIPTION
**Pull request by anjeylink**

### CODE REVIEW

**Review by vuxorpadd — APPROVED** [review]

File: Classes/Importer/JobDataImporter.php [hunk 1]
```diff
@@ -165,6 +182,32 @@ public function import(array $importData, int $pid)
...
+    {
+        $db = $GLOBALS['TYPO3_DB'];
+
+        $result = $db->sql_query(
+            'SELECT uniqalias.uid, url_cache_id FROM ' .
+            'tx_realurl_uniqalias uniqalias JOIN tx_pxaintelliplanjobs_domain_model_job jobs ON uniqalias.value_id=jobs.uid ' .
+            'LEFT JOIN tx_realurl_uniqalias_cache_map cache ON uniqalias.uid=cache.alias_uid ' .
+            'WHERE jobs.deleted=1 AND uniqalias.tablename=' . $db->fullQuoteStr('tx_pxaintelliplanjobs_domain_model_job', 'tx_realurl_uniqalias'));
+
+        while (false !== ($data = $db->sql_fetch_assoc($result))) {
```
Comment (MattiasNilsson) [inline Classes/Importer/JobDataImporter.php]: No Yoda conditions!

File: Classes/Importer/JobDataImporter.php [hunk 2]
```diff
@@ -165,6 +182,32 @@ public function import(array $importData, int $pid)
...
+    /**
+     * Clean all unique aliases for deleted jobs
+     */
+    protected function cleanUpUniqueAliasForDeletedRecords()
+    {
+        $db = $GLOBALS['TYPO3_DB'];
+
+        $result = $db->sql_query(
+            'SELECT uniqalias.uid, url_cache_id FROM ' .
+            'tx_realurl_uniqalias uniqalias JOIN tx_pxaintelliplanjobs_domain_model_job jobs ON uniqalias.value_id=jobs.uid ' .
```
Comment (MattiasNilsson) [inline Classes/Importer/JobDataImporter.php]: `uniqalias.value_id = jobs.uid`

File: Classes/Importer/JobDataImporter.php [hunk 3]
```diff
@@ -165,6 +182,32 @@ public function import(array $importData, int $pid)
...
+            'LEFT JOIN tx_realurl_uniqalias_cache_map cache ON uniqalias.uid=cache.alias_uid ' .
```
Comment (MattiasNilsson) [inline Classes/Importer/JobDataImporter.php]: `uniqalias.uid = cache.alias_uid`

File: Classes/Importer/JobDataImporter.php (hunk 1 above)
Comment (vuxorpadd) [reply Classes/Importer/JobDataImporter.php]: Yoda rules!

**Pull request merged**